### PR TITLE
Updated b-tag candidate kinematics

### PR DESCRIPTION
**Pull request by ferencek**
This PR introduces updated handling of candidate kinematics in the b-tagging code with the goal of improving the agreement between discriminators computed from AOD and MiniAOD (problem reported, for example, in https://hypernews.cern.ch/HyperNews/CMS/get/btag/1232.html).

In AOD PF candidates and tracks linked to them have their own separate momenta defined while in MiniAOD PackedCandidates store track info with the track momentum set to the candidate momentum in order to save some space.  The main change introduced by this PR is the use of the PF candidate momentum instead of the track momentum where possible. This change results in a better agreement between the b-tag discriminators computed from AOD and those computed from MiniAOD. The improved agreement can be seen in the plots attached below.

**MiniAOD vs AOD CVSv2IVF discriminator before the fix:**
miniaod_vs_aod_beforefix

**MiniAOD vs AOD CVSv2IVF discriminator after the fix:**
miniaod_vs_aod_afterfix

The introduced changes leave the overall b-tagging performance essentially unchanged, as can be seen from the plots attached below (before: red circles, after: blue triangles).

**CVSv2IVF performance against light jets:**
flaveffvsbeff_dusg_pfcsvv2ivf

**CVSv2IVF performance against c jets:**
flaveffvsbeff_c_pfcsvv2ivf

### CODE REVIEW

File: RecoBTag/SecondaryVertex/src/TrackSelector.cc [hunk 1]
```diff
@@ -71,16 +69,49 @@ TrackSelector::operator () (const Track &track,
   }
   else  jtaPassed = true;
 
+  return track.pt() >= minPt &&
+    VectorUtil::DeltaR(jet.momentum(),
```
Comment (slava77) [inline RecoBTag/SecondaryVertex/src/TrackSelector.cc]: please use reco::deltaR2 (CMSSW and no sqrt), this will need maxJetDeltaR2

Comment (slava77) [inline RecoBTag/SecondaryVertex/src/TrackSelector.cc]: more specifically,
```reco::deltaR2(jet, track)```

Comment (ferencek) [inline RecoBTag/SecondaryVertex/src/TrackSelector.cc]: @slava77, this is now implemented. Sorry for the delay. There was a problem with AFS on lxplus which made my working area inaccessible.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/12984	95acaee
- changes are in line with the description (most notably, the "use of the PF candidate momentum instead of the track momentum where possible")
- jenkins tests pass and comparisons with the baseline show small differences only in b-tag related variables, as expected
    - among the taggers monitored in fwlite (easier to get a list of branches modified), values change in pfCombinedCvsBJetTags, pfCombinedCvsLJetTags, pfCombinedInclusiveSecondaryVertexV2BJetTags, pfCombinedMVABJetTags, pfCombinedMVAV2BJetTags. This is in line with the code changes.

**Comment by davidlange6** [comment]
+1

**Pull request merged**